runtime/cudaq/platform/default/rest_server/RemoteRESTQPU.cpp

```diff
@@ -0,0 +1,365 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "common/RuntimeMLIR.h"
+#include "cudaq/Frontend/nvqpp/AttributeNames.h"
+#include "cudaq/Optimizer/CodeGen/OpenQASMEmitter.h"
+#include "cudaq/Optimizer/CodeGen/Passes.h"
+#include "cudaq/Optimizer/Dialect/CC/CCDialect.h"
+#include "cudaq/Optimizer/Dialect/Quake/QuakeDialect.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
+
+#include "llvm/Bitcode/BitcodeReader.h"
+#include "llvm/Bitcode/BitcodeWriter.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IRReader/IRReader.h"
+#include "llvm/Support/Base64.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/ExecutionEngine/ExecutionEngine.h"
+#include "mlir/ExecutionEngine/OptUtils.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Pass/PassRegistry.h"
+
+#include "common/ExecutionContext.h"
+#include "common/Logger.h"
+#include "common/NoiseModel.h"
+#include "common/RestClient.h"
+#include "cudaq.h"
+#include "cudaq/platform/qpu.h"
+#include "cudaq/platform/quantum_platform.h"
+#include "cudaq/qis/qubit_qis.h"
+#include "cudaq/spin_op.h"
+#include <fstream>
+#include <iostream>
+#include <spdlog/cfg/env.h>
+
+#include "JsonConvert.h"
+#include "nvqir/CircuitSimulator.h"
+#include <arpa/inet.h>
+#include <execinfo.h>
+#include <signal.h>
+#include <sys/socket.h>
+
+namespace {
+using namespace mlir;
+class RemoteQpuExecutionQueue : public cudaq::QuantumExecutionQueue {
+public:
+  std::thread::id getExecutionThreadId() const { return thread.get_id(); }
```

Comment:
should / can this just be added to QuantumExecutionQueue?

Reply:
Yea, we should. Fixed.